yaptide/batch/batch_methods.py

```diff
@@ -327,3 +327,27 @@ def delete_job(simulation: BatchSimulationModel, user: KeycloakUserModel,
         return {"message": "Job cancelation failed"}, 500
 
     return {"message": "Job canceled"}, 200
+
+
+def fetch_results_without_stopping(simulation: BatchSimulationModel, user: KeycloakUserModel,
+                                   cluster: ClusterModel) -> tuple[dict, int]:  # skipcq: PYL-W0613
+    """Function that stops simulation without cancelling collection job"""
+    array_id = simulation.array_id
+    tasks = fetch_celery_tasks_by_sim_id(sim_id=simulation.id)
+    try:
+        con = get_connection(user=user, cluster=cluster)
+        i = 1
+        fetch_str = 'scancel --signal=SIGINT'
+        delete_str = 'scancel'
+        for task in tasks:
+            if task.task_state == EntityState.RUNNING.value:
+                fetch_str += f' {array_id}_{i}'
+            if task.task_state in (EntityState.PENDING.value, EntityState.UNKNOWN.value):
+                delete_str += f' {array_id}_{i}'
```

Comment:
add some comments what happens here, explaining that you will with SIGINT only running tasks, asking them to dump the results and kill in normal way all pending tasks